torch/_inductor/kernel/conv.py

```diff
@@ -363,6 +365,13 @@ def channels_last_conv():
         and not transposed
         and is_zeros(output_padding)
         and groups == 1
+        # We tried free_unbacked_symbols first but inductor somehow get
+        # unbacked symbols having 's' as prefix and those are not
+        # counted as unbacked symbols by free_unbacked_sybmols function.
+        #
+        # Call free_symbols instead. The downside is for some cases, we
+        # conservatively not converting conv1x1 to mm. Not a big deal.
```

Comment:
Because these are backed symints, you have another option, which is to (1) check the hint and (2) add a guard. Also, in both backed and unbacked case, you can check if it's statically known that the numel is > 0. In particular, because of 0/1 specialization, you will nearly always know that the dynamic shapes in question are > 0 and therefore the optimization should apply. So this is basically free (no extra guards) and I would greatly prefer to preserve this optimization even with dynamic shapes.

Reply:
I can keep this optimization for dynamic shapes. But there is still one thing puzzles me.

From the symbol prefix ('s' here), we know they are backed symint. But those symbols are the number of non-zeros from some integer tensors in my unit test. They are data dependent and seems more like unbacked symint. Any idea what's going on here?